client/elastic-client.go

```diff
@@ -0,0 +1,130 @@
+// package client provides a new ES Client and CRUD operations for it. Package uses olivere/elastic client.
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/olivere/elastic"
+)
+
+const (
+	IndexName = "files"
+)
+
+type File struct {
+	Name        string `json:"name"`
+	Url         string `json:"url"`
+	Description string `json:"description"`
+}
+// NewElasticClient provides a new elastic.Client allocation and creation of the Index in ES.
+//
+// Returns a pointer to the created Client and an error
+func NewElasticClient() (*elastic.Client, error) {
+	client, err := elastic.NewClient(elastic.SetURL("http://elastic:9200"))
+	if err != nil {
+		return client, err
+	}
+
+	//check if Index hasn't been created yet
+	exists, err := client.IndexExists(IndexName).Do(context.Background())
+	if err != nil {
+		return client, err
+	}
+	if !exists {
+		_, err = client.CreateIndex(IndexName).Do(context.Background())
+	}
+
+	return client, err
+}
+
+// CreateDoc provides a doc creation in the Index using elastic.Client
+//
+// Returns an Id of the doc and an error
+func CreateDoc(client *elastic.Client, file File) (string , error) {
```

Comment:
Why is it better to use embedding instead of passing `*elastic.Client` pointer to the functions? While I was writing this piece of code, I had a few ways to create the CRUD operations. At first, I tried to use alias for `*elastic.Client`, but it was unsuccessful. Then, I decided to create these functions the same way, I used to create it in Python.

Reply:
Try to think about another backend storage, e.g. postgres. How you would add this? What about SOLID Open/Closed principle?

Reply:
It seems like I should read about OOP in Golang because I thought it's, roughly speaking, almost functional. Then, you've got a point here -- I'll add the client embedding. Thank you!